frontend/src/pages/RunDetails.tsx

```diff
@@ -877,30 +905,52 @@ class RunDetails extends Page<RunDetailsInternalProps, RunDetailsState> {
 
   private async _loadSelectedNodeLogs(): Promise<void> {
     const selectedNodeDetails = this.state.selectedNodeDetails;
-    if (!selectedNodeDetails) {
+    const namespace = this.state.workflow?.metadata?.namespace;
+    if (!selectedNodeDetails || !namespace) {
       return;
     }
     this.setStateSafe({ sidepanelBusy: true });
+
+    let logsBannerMessage = '';
+    let logsBannerAdditionalInfo = '';
+    let logsBannerMode = '' as Mode;
+
     try {
-      const logs = await Apis.getPodLogs(
-        selectedNodeDetails.id,
-        this.state.workflow?.metadata?.namespace,
-      );
-      selectedNodeDetails.logs = logs;
+      selectedNodeDetails.logs = await Apis.getPodLogs(selectedNodeDetails.id, namespace);
+    } catch (err) {
+      let errMsg = await errorToMessage(err);
+      logsBannerMessage = 'Failed to retrieve pod logs.';
+
+      // if pod can't be found, then assume it has gone from the server
+      if (await this.podNotFound(selectedNodeDetails.id, namespace)) {
```

Comment:
nit: just confirm the error message returned from `Apis.getPodLogs` has no hint whether the pod is not found?
If no, this seems a reasonable workaround.

Reply:
it does allude to workflow not being found, or something along those lines, but the error code is 500, so would have to do a msg contains text type check. Feels a bit brittle but happy to do that instead

Reply:
I'm okay with text check in this PR, if you feel you have capacity, we can also detect and change the error code in https://github.com/kubeflow/pipelines/blob/57f415d2d1960e4017d7eb05b0fea85d21436add/frontend/server/handlers/pod-logs.ts#L76

Reply:
ohh I thought this was interfacing straight with argo 🤦 . This makes sense then yeah , will do here

Reply:
so I've updated this but don't have time to stand up the whole backend to test it. is there some kind of short cut way to do it?

Reply:
Do you have a running KFP cluster?
If yes, you can follow https://github.com/kubeflow/pipelines/blob/master/frontend/README.md#proxy-to-a-real-cluster, you will need to develop "Client UI + Node server". It will send api requests to the cluster for anything else.

Reply:
unfortunately, we haven't added integration tests for pod logs api method.
https://github.com/kubeflow/pipelines/blob/d6920ca2ad0c36c195c8f483f4f1ac27d4b7e82f/frontend/server/app.test.ts#L1041-L1042

So I'd recommend just checking text match in this PR and you are welcome to change that implementation to node server status code in a separate one, preferably with test coverage.